tests/functional/s3curl/tests.js

```diff
@@ -124,370 +127,570 @@ function provideRawOutput(args, cb) {
-        provideRawOutput(['--',
-        `${endpoint}/${bucket}?` +
-        `&prefix=${prefix}`, '-v'], (httpCode, rawOutput) => {
-            assert.strictEqual(httpCode, '200 OK');
-            parseString(rawOutput.stdout, (err, result) => {
-                if (err) {
-                    assert.ifError(err);
-                }
-                const keys = Object.keys(result.ListBucketResult);
-                const location = keys.indexOf('CommonPrefixes');
-                assert.strictEqual(location, -1);
-                assert.strictEqual(result.ListBucketResult
-                    .Contents[0].Key[0], 'topLevel/test1MB1');
-                done();
+        provideRawOutput(
+            ['--', `${bucketPath}?&prefix=${prefix}`, '-v'],
+            (httpCode, rawOutput) => {
+                assert.strictEqual(httpCode, '200 OK');
+                parseString(rawOutput.stdout, (err, result) => {
+                    if (err) {
+                        assert.ifError(err);
+                    }
+                    const keys = Object.keys(result.ListBucketResult);
+                    const location = keys.indexOf('CommonPrefixes');
+                    assert.strictEqual(location, -1);
+                    assert.strictEqual(result.ListBucketResult
+                        .Contents[0].Key[0], `${basePath}${upload}1`);
+                    done();
+                });
             });
-        });
     });
 
     it('should provide a next marker if maxs keys exceeded ' +
         'and delimiter specified', done => {
-        provideRawOutput(['--',
-        `${endpoint}/${bucket}?` +
-        'delimiter=x&max-keys=2', '-v'], (httpCode, rawOutput) => {
-            assert.strictEqual(httpCode, '200 OK');
-            parseString(rawOutput.stdout, (err, result) => {
-                if (err) {
-                    assert.ifError(err);
-                }
-                assert.strictEqual(result.ListBucketResult
-                    .NextMarker[0], 'topLevel/test1MB2');
-                assert.strictEqual(result.ListBucketResult
-                    .IsTruncated[0], 'true');
-                done();
+        provideRawOutput(
+            ['--', `${bucketPath}?delimiter=x&max-keys=2`, '-v'],
+            (httpCode, rawOutput) => {
+                assert.strictEqual(httpCode, '200 OK');
+                parseString(rawOutput.stdout, (err, result) => {
+                    if (err) {
+                        assert.ifError(err);
+                    }
+                    assert.strictEqual(result.ListBucketResult
+                        .NextMarker[0], `${basePath}${upload}2`);
+                    assert.strictEqual(result.ListBucketResult
+                        .IsTruncated[0], 'true');
+                    done();
+                });
             });
-        });
     });
 });
 
 describe('s3curl head bucket', () => {
+    before('create file to put', done => {
```

Comment:
Ah, that reminds me that I didn't check the "before" and "after" messages. I guess I'll have to update this. Thanks